### PR TITLE
Module and notebook for the pipeline (CMFGEN to HDF5)

### PR DESCRIPTION
**Pull request by epassaro**
- [x] New module with `h5dump` function.
- [x] Use `logger` instead
- [x] Notebook for the pipeline
https://nbviewer.jupyter.org/github/epassaro/carsus/blob/output-notebook/docs/io/cmfgen/pipeline.ipynb?flush_cache=true
- [x] Remove try/exception block?
- [x] Add ignore pattern list (suggestion) 
- [x] Docstring
- [x] Ready to merge

### CODE REVIEW

File: carsus/io/cmfgen/hdfgen.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    for chunk in files_chunked:
+
+        _ = []
+        for fname in chunk:
+            try:
+                obj = parser.__class__(fname)
+                logger.info('Parsed {}'.format(fname.replace(cmfgen_dir + 'atomic/', '')))
+                _.append(obj)
+
+            except:
```
Comment (wkerzendorf) [inline carsus/io/cmfgen/hdfgen.py]: well what errors do you expect?

File: carsus/io/cmfgen/hdfgen.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import os
+import glob
+from carsus import logger
+
+
+def h5dump(cmfgen_dir, patterns, parser, chunk_size=10):
```
Comment (wkerzendorf) [inline carsus/io/cmfgen/hdfgen.py]: call it `hdf_dump`

File: carsus/io/cmfgen/hdfgen.py (hunk 1 above)
Comment (epassaro) [reply carsus/io/cmfgen/hdfgen.py]: I think my problem is how to deal with errors I don't expect. Most of situations are contemplated in try/except blocks in `base.py`.

We have two 'kind' of parsers: 

- **energy levels, oscillator and collisional strengths:** Pandas-based. In the end we expect some Pandas error.

- **photoionization cross section:** handcrafted with Python standard library, and stored in Pandas dataframes. 


Then there are the exceptions related to `to_hdf` method.

File: carsus/io/cmfgen/hdfgen.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import glob
+import logging
+
+logger = logging.getLogger(__name__)
+
+def hdf_dump(cmfgen_dir, patterns, parser, chunk_size=10, ignore_patterns=[]):
+    """Function to parse and dump the entire CMFGEN database.
+
+    Arguments:
```
Comment (wkerzendorf) [inline carsus/io/cmfgen/hdfgen.py]: this needs to be `numpydoc` style.

Comment (epassaro) [reply carsus/io/cmfgen/hdfgen.py]: Oh, I did it with the VSCode plugin, I'll try to configure it.

**Pull request merged**